packages/msgraph/test/Adaptor.test.js

```diff
@@ -37,68 +39,439 @@ describe('execute', () => {
-      data: {
-        fullName: 'Mamadou',
-        gender: 'M',
+    };
+
+    const finalState = await execute(
+      getDrive({ id: 'b!YXzpkoLwR06bxC8tNdg71m_' }, 'mydrive', state => {
+        // write the drives object back to state before it gets cleaned up
+        state.result = state.drives;
+        return state;
+      })
+    )(state);
+
+    expect(finalState.result.mydrive).to.eql(fixtures.driveResponse);
+  });
+
+  it('should get the default drive for a site', async () => {
+    const state = {
+      configuration: {
+        accessToken: fixtures.accessToken,
       },
     };
 
     const finalState = await execute(
-      create('patients', {
-        name: dataValue('fullName')(state),
-        gender: dataValue('gender')(state),
+      getDrive(
+        { id: 'openfn.sharepoint.com', owner: 'sites' },
+        undefined,
+        state => {
+          // write the drives object back to state before it gets cleaned up
+          state.result = state.drives;
+          return state;
+        }
+      )
+    )(state);
+
+    expect(finalState.result.default).to.eql(fixtures.driveResponse);
+  });
+  it('should throws 400 error', async () => {
+    const state = {
+      configuration: {
+        accessToken: fixtures.accessToken,
+      },
+    };
+
+    await execute(
+      getDrive({ id: 'noAccess', owner: 'sites' })(state).catch(e => {
+        expect(e.message).to.contain(
+          fixtures.invalidRequestResponse.error.message
+        );
       })
     )(state);
+  });
 
-    expect(finalState.data).to.eql({
-      fullName: 'Mamadou',
-      gender: 'M',
-      id: 7,
-    });
+  it('throws 401 error with invalidToken message', async () => {
+    const state = {
+      configuration: {
+        accessToken: fixtures.invalidToken,
+      },
+    };
+
+    await execute(
+      getDrive({ id: 'openfn.sharepoint.com', owner: 'sites' })(state).catch(
+        e => {
+          expect(e.message).to.contain(
+            fixtures.invalidTokenResponse.error.message
+          );
+        }
+      )
+    )(state);
+  });
+
+  it('should throws 401 error with expiredToken message', async () => {
+    const state = {
+      configuration: {
+        accessToken: fixtures.expiredToken,
+      },
+    };
+
+    await execute(
+      getDrive({ id: 'openfn.sharepoint.com', owner: 'sites' })(state).catch(
+        e => {
+          expect(e.message).to.contain(
+            fixtures.expiredTokenResponse.error.message
+          );
+        }
+      )
+    )(state);
+  });
+});
+
+describe('getFolder', () => {
+  it('should get a folder metadata by id', async () => {
+    const state = {
+      configuration: {
+        accessToken: fixtures.accessToken,
+      },
+      drives: {
+        default: {
+          id: 'b!YXzpkoLwR06bxC8tNdg71m_',
+        },
+      },
+    };
+
+    const finalState = await execute(
+      getFolder(
+        '01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3',
+        { metadata: true },
+        state => {
```

Comment:
I'm being picky but we don't actually need to use the callback here - let's remove it as it only confuses the test!

Reply:
Done [c7d7243](https://github.com/OpenFn/adaptors/pull/297/commits/c7d72434d44e189403c6b983dfef73da42c21d6d)